dvc/stage.py

```diff
@@ -164,10 +168,23 @@ def __init__(
         self.outs = outs
         self.deps = deps
         self.md5 = md5
+        self.sha256 = sha256
         self.locked = locked
         self.tag = tag
         self._state = state or {}
 
+        if repo and repo.config and repo.config.config:
```

Comment:
this code matches the one in remote local. How about we just use `repo.cache.local.hash` instead, since it is pretty much the same thing and local cache is guaranteed to exist?